reconstruct/reconstruct.go

```diff
@@ -93,78 +93,41 @@ func Yhat(r []float64) int {
 	return pos
 }
 
-//get the nodes of the entire tree in a slice
-func Get_tree(node *tree_newick.NTree) []*tree_newick.NTree {
-	var branch []*tree_newick.NTree
-	branch = append(branch, node)
-	if node.Right != nil {
-		b := Get_tree(node.Right)
-		branch = append(branch, b...)
-	}
-	if node.Left != nil {
-		a := Get_tree(node.Left)
-		branch = append(branch, a...)
-	}
-	return branch
-}
-
-//get the interior nodes of the tree
-func Get_branch(node *tree_newick.NTree) []*tree_newick.NTree {
-	var branch []*tree_newick.NTree
-	if node.Left != nil && node.Right != nil {
-		branch = append(branch, node)
-		a := Get_branch(node.Left)
-		b := Get_branch(node.Right)
-		branch = append(branch, a...)
-		branch = append(branch, b...)
-	}
-	return branch
-}
-
-//get the leafs of the tree
-func Get_leaf(node *tree_newick.NTree) []*tree_newick.NTree {
-	var leaf []*tree_newick.NTree
-	if node.Left != nil && node.Right != nil {
-		a := Get_leaf(node.Left)
-		b := Get_leaf(node.Right)
-		leaf = append(leaf, a...)
-		leaf = append(leaf, b...)
-	}
-	if node.Left == nil && node.Right == nil {
-		leaf = append(leaf, node)
-	}
-	return leaf
-}
-
-//set the state of the tree given the Fasta and the position
-func Set_state(node *tree_newick.NTree, pos int) {
-	leaf := Get_leaf(node)
-	var leaf_names []string
+//set the state of the tree given the Fasta and the position (zero-based)
+func SetState(node *expandedTree.ETree, pos int) {
+	leaf := expandedTree.GetLeaf(node)
```

Comment:
From looking later in the function, is the only reason to get the leaves to know if this node we are currently at is a leaf?  Can we just not calculate this and instead use "if node.Right == nil && node.Left == nil" to test if we are currently at a leaf later in the function?